sync/sync_test.go

```diff
@@ -94,42 +92,8 @@ func TestSyncBlocks(t *testing.T) {
 
 		mockSNData := mocks.NewMockStarknetData(mockCtrl)
 
-		syncingHeight := uint64(0)
-
-		mockSNData.EXPECT().BlockByNumber(gomock.Any(), gomock.Any()).DoAndReturn(func(_ context.Context, height uint64) (*core.Block, error) {
-			curHeight := atomic.LoadUint64(&syncingHeight)
-			// reject any other requests
-			if height != curHeight {
-				return nil, errors.New("try again")
-			}
-			return gw.BlockByNumber(context.Background(), syncingHeight)
-		}).AnyTimes()
-
-		reqCount := 0
-		mockSNData.EXPECT().StateUpdate(gomock.Any(), gomock.Any()).DoAndReturn(func(_ context.Context, height uint64) (*core.StateUpdate, error) {
-			curHeight := atomic.LoadUint64(&syncingHeight)
-			// reject any other requests
-			if height != curHeight {
-				return nil, errors.New("try again")
-			}
-
-			reqCount++
-			ret, err := gw.StateUpdate(context.Background(), curHeight)
-			require.NoError(t, err)
-
-			switch reqCount {
-			case 1:
-				return nil, errors.New("try again")
-			case 2:
-				ret.BlockHash = new(felt.Felt) // fail sanity checks
-			case 3:
-				ret.OldRoot = new(felt.Felt).SetUint64(1) // fail store
-			default:
-				reqCount = 0
-				atomic.AddUint64(&syncingHeight, 1)
-			}
-
-			return ret, nil
+		mockSNData.EXPECT().StateUpdateWithBlock(gomock.Any(), gomock.Any()).DoAndReturn(func(_ context.Context, height uint64) (*core.StateUpdate, *core.Block, error) {
```

Comment:
I think we lost test coverage here. In the old implementation, we have multiple cases that exercise different code paths: failing sanity checks, failing store, etc. Can we get that coverage back?

Reply:
Ah.. true, missed that.

Reply:
@Brivan-26 any updates on this?

Reply:
@omerfirmak Working no adding the coverage back (getting annoying issue when using the new endpoint and trying to solve it).